cmake/modules/hunter_check_toolchain_for_define.cmake

```diff
@@ -0,0 +1,36 @@
+# Copyright (c) 2016 NeroBurner
+# All rights reserved.
+
+include(hunter_test_string_not_empty)
+
+function(hunter_check_toolchain_for_define definition output_var)
+  # the toolchain path must be set
+  hunter_test_string_not_empty("${HUNTER_TOOLCHAIN_ID_PATH}")
+  set(TOOLCHAIN_INFO_FILE "${HUNTER_TOOLCHAIN_ID_PATH}/toolchain.info")
+
+  # read toolchain.info file into memory
+  file(READ "${TOOLCHAIN_INFO_FILE}" toolchain_contents)
+
+  # Convert file contents into a CMake list (where each element in the list
+  # is one line of the file)
+  string(REGEX REPLACE ";" "\\\\;" toolchain_contents "${toolchain_contents}")
+  string(REGEX REPLACE "\n" ";"    toolchain_contents "${toolchain_contents}")
+
+  set(definition_found 0)
+  foreach(line ${toolchain_contents})
+    # trim line
+    string(REGEX REPLACE "^#define " "" line "${line}")
```

Comment:
With the note below I think we may have some false positives here for lines which don't start with `#define`. I think next regex should be check and analyzed:
```cmake
string(REGEX MATCH "^#define ${definition} .*$" match "${line}")
```